File: src/data/loader.ts
import type { AppData } from '../types.js';
import { loadSkills } from './skills.js';
import { loadHooks } from './hooks.js';
import { loadModels } from './models.js';
import { loadWorkspace } from './workspace.js';
import { loadMcp } from './mcp.js';
import { loadSessions } from './sessions.js';
import { loadCron } from './cron.js';
import { loadMemory } from './memory.js';
import { loadUpdates } from './updates.js';
import { loadWebhooks } from './webhooks.js';
import { loadAuditLog } from './auditlog.js';

export function loadAll(): AppData {
  return {
    skills:    loadSkills(),
    hooks:     loadHooks(),
    models:    loadModels(),
    workspace: loadWorkspace(),
    mcp:       loadMcp(),
    sessions:  loadSessions(),
    cron:      loadCron(),
    memory:    loadMemory(),
    updates:   loadUpdates(),
    webhooks:  loadWebhooks(),
    auditlog:  loadAuditLog(),
  };
}

File: internal/data/loader.go
package data

func LoadAll() AppData {
	return AppData{
		Skills:      LoadSkills(),
		Hooks:       LoadHooks(),
		Models:      LoadModels(),
		Workspace:   LoadWorkspace(),
		MCP:         LoadMCP(),
		Sessions:    LoadSessions(),
		Cron:        LoadCron(),
		Tasks:       LoadTasks(),
		Memory:      LoadMemory(),
		Updates:     LoadUpdates(),
		Webhooks:    LoadWebhooks(),
		AuditLog:    LoadAuditLog(),
		AgentConfig: LoadAgentConfig(),
		Devices:     LoadDevices(),
		Logs:        LoadLogs(),
	}
}
